src/components/pages/notifier/buy/NotifierOffersPage.tsx

```diff
@@ -87,43 +90,71 @@ const NotifierOffersPage: FC = () => {
   }
 
   const [selectedProvider, setSelectedProvider] = useState<ProviderItem>()
+  const [collection, setCollection] = useState<MarketplaceItem[]>([])
 
-  const providers = Array.from(new Set(items.map(({ provider }) => provider)))
-
-  const collection: MarketplaceItem[] = providers
-    .map<MarketplaceItem>((provider) => {
-      const providerPlans = items.filter((item) => item.provider === provider)
-
-      const { priceFiatRange, ...offerDetails } = mapPlansToOffers(
-        providerPlans, crypto,
-      )
-
-      const isSelected = selectedProvider?.id === provider
-
-      return {
-        id: provider,
-        provider: <AddressItem value={provider} />,
-        ...offerDetails,
-        priceFiatRange: (
-          <ItemWUnit
-            type="mediumPrimary"
-            value={priceFiatRange}
-            unit={currentFiat}
-          />
-        ),
-        action1: account === provider ? 'your offer' : (
-          <SelectRowButton
-            id={provider}
-            isSelected={isSelected}
-            handleSelect={(): void => {
-              setSelectedProvider(isSelected
-                ? undefined
-                : { id: provider, plans: providerPlans })
-            }}
-          />
-        ),
-      }
-    })
+  useEffect(() => {
+    if (items?.length) {
+      const providers = Array.from(new Set(items
+        .map(({ provider }) => provider)))
+
+      Promise.all(providers
+        .map<Promise<MarketplaceItem>>(async (provider) => {
+          const providerPlans = items
+            .filter((item) => item.provider === provider)
+          const { url } = providerPlans[0]
+
+          const notifierService = new SubscriptionPlans(url)
+          notifierService.connect(Logger.getInstance().debug)
+          const hasActivePlans = await notifierService.hasActivePlans()
+
+          const { priceFiatRange, ...offerDetails } = mapPlansToOffers(
+            providerPlans, crypto,
+          )
+
+          const isSelected = selectedProvider?.id === provider
+
+          const selectButton = (
+            <SelectRowButton
+              disabled={!hasActivePlans}
+              id={provider}
+              isSelected={isSelected}
+              handleSelect={(): void => {
+                setSelectedProvider(isSelected
+                  ? undefined
+                  : { id: provider, plans: providerPlans })
+              }}
+            />
+          )
+          let action1 = selectButton
+
+          if (!hasActivePlans) {
+            action1 = (
+              <Tooltip title="This provider doesn't have any active plan at the moment">
+                <span>
+                  {selectButton}
+                </span>
+              </Tooltip>
+            )
+          }
```

Comment:
Mutation isn't nice thou...why not just
```suggestion
          const action1 = hasActivePlans
            ? selectButton
            :  (
              <Tooltip title="This provider doesn't have any active plan at the moment">
                <span>
                  {selectButton}
                </span>
              </Tooltip>
            )
```
?

Reply:
Done, thanks!